### DISCUSSION TITLE
labels of left and right or xmin and xmax

### DISCUSSION THREAD

**Discussion by Bahar527** [PMEAL/OpenPNM#2755] [Q&A]
hi,
I have extracted a network from a 3D CT image of porous media using snow2 algorithm in porespy and I have imported it into openPNM by op.io.network_from_porespy module. My problem is that there is no left and right or xmin and xmax label in my network(or top and down).  So I can not use many modules in openPNM specifically for predicting relative permeability. Can you help me with this issue?
I'm looking forward to hearing from you.

**Comment by likeduck** [comment]
Hi, I am learning the same things as you. Hope my answer will help you.

First, find a possible coordinate range.
`Ps = pn['pore.coords'][:, 0] < 7.13e-7  # means X-coords in [0, 7.13e-7]; 0 represents X, 1 represents Y and 2 represents Z`
Next, define them.
`pn.set_label(label='pore.left', pores=Ps)`
If you want the connected throats, like this:
```
Ts = pn.find_neighbor_throats(pores=pn['pore.left'], mode='or')
pn.set_label(label='yourname', throats=Ts)
```
The above code is very likely feasible because it is the manuscript I have run. You can try!
